### PR TITLE
feat: Schedule: Implement "Add to calendar" for a complete schedule

### LINKED ISSUE #5760: Schedule: Implement "Add to calendar" for a complete schedule
We already have the feature "Add to calendar" for the event times. Please also implement "Add to calendar" for events that have a dedicated schedule with sessions.

If the event has sessions with a schedule show two buttons instead of one in the "Add to Calendar" area on the right hand side:
1. On the left show a button with a calendar icon and "Event" (which is using the current implementation) as follows: 📅 Add Event
2. One the right show a button with a calendar icon and "Sessions" as follows: 📅 Add Sessions

Note: If the event does not have sessions, please show only one large button.

Screenshot from 2020-11-25 20-37-35

### PR DESCRIPTION
**Pull request by Draco9421**
Screenshot from 2021-02-11 20-09-08
Screenshot from 2021-02-11 20-09-12

Fixes #5760

#### Changes proposed in this pull request:

-
-
-

#### Checklist

- [ ] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [ ] My branch is up-to-date with the Upstream `development` branch.
- [ ] The acceptance, integration, unit tests and linter pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by daretobedifferent18** [comment]
@Draco9421 Pls mention which issue are you solving

**Review by daretobedifferent18 — COMMENTED** [review]
Acc to issue, if event does not have session, then only show one large button
but currently showing bith buttons 
Compare:

File: app/components/public/add-to-calender.hbs [hunk 1]
```diff
@@ -9,6 +9,38 @@
       {{/if}}
       {{general-date @event.startsAt 'tz' tz=@event.timezone}}
     </div>
+
+    {{#if this.Session}}
```
Comment (daretobedifferent18) [inline app/components/public/add-to-calender.hbs:13]: instead use ```{{#if @event.sessions}}```

**Comment by mariobehling** [comment]
- Please also add calendar icon.
- Add Events -> Add Event

File: app/components/public/add-to-calender.ts [hunk 2]
```diff
@@ -27,6 +27,10 @@ export default class AddToCalender extends Component<Args> {
     return this.startsAt.isSame(this.endsAt, 'day');
   }
 
+  get Session(): boolean {
+    return this.args.event.sessions;
+  }
+
```
[the remark this answers is not in the document]
Comment (Draco9421) [reply app/components/public/add-to-calender.ts:33]: I didn't know that we could check sessions by {{#if @event.sessions}}

File: app/components/public/add-to-calender.hbs (hunk 1 above)
Comment (Draco9421) [reply app/components/public/add-to-calender.hbs:13]: ok sir

**Comment by Draco9421** [comment]
Screenshot from 2020-12-20 15-23-07

Does this look ok sir ?

File: app/components/public/add-to-calender.hbs [hunk 3]
```diff
@@ -19,5 +51,6 @@
         {{/each}}
       </div>
     </UiDropdown>
+    {{/if}}
   </div>
-</div>
+</div>
```
[the remark this answers is not in the document]
Comment (daretobedifferent18) [reply app/components/public/add-to-calender.hbs:59]: Pls add line at end of file

**Review by daretobedifferent18 — COMMENTED** [review]
Currenctly it is standing out
Screenshot_2020-12-20 UNESCO TechCul Ideathon eventyay

for this pls add padding as was before for Add to Calender

**Review by daretobedifferent18 — COMMENTED** [review]
Before | After 
--------- | ---------
Screenshot_2020-12-20 UNESCO TechCul Ideathon eventyay(2) | Screenshot_2020-12-20 UNESCO TechCul Ideathon eventyay(1)

There is should not be a line below. Pls fix this as well

File: app/components/public/add-to-calender.hbs [hunk 4]
```diff
@@ -9,15 +9,50 @@
...
-    <UiDropdown class="ui floating dropdown button mt-4">
-      {{t 'Add to Calendar'}}
-      <div class="menu">
-        {{#each this.calendarUrls as |calendar|}}
-          <a href={{calendar.url}} class="item" target="_blank" rel="noreferrer nofollow">
-            {{calendar.name}}
-          </a>
-        {{/each}}
+
+    {{#if @event.sessions}}
```
Comment (daretobedifferent18) [inline app/components/public/add-to-calender.hbs:13]: fix the indentation

**Comment by Draco9421** [comment]
why it is giving error ?

**Review by daretobedifferent18 — APPROVED** [review]
its working fine for me

**Comment by Draco9421** [comment]
> its working fine for me

No sir I was talking about this CI / Test (pull_request).

**Comment by daretobedifferent18** [comment]
> > its working fine for me
> 
> No sir I was talking about this CI / Test (pull_request).

I have commented this for PR

File: app/components/public/add-to-calender.hbs [hunk 5]
```diff
@@ -9,15 +9,50 @@
...
+                  </a>
+                {{/each}}
+              </div>
+            </UiDropdown>
+          </div>
+        </div>
       </div>
-    </UiDropdown>
+    {{else}}
+      <UiDropdown class="ui floating dropdown button mt-4">
```
Comment (daretobedifferent18) [inline app/components/public/add-to-calender.hbs:45]: here is code duplication. If that could be handle pls do that

File: app/components/public/add-to-calender.hbs [hunk 6]
```diff
@@ -9,15 +9,43 @@
...
-      <div class="menu">
-        {{#each this.calendarUrls as |calendar|}}
-          <a href={{calendar.url}} class="item" target="_blank" rel="noreferrer nofollow">
-            {{calendar.name}}
-          </a>
-        {{/each}}
+
+    <div class="ui grid">
+      <div class="sixteen column row">
+        <div class="coloum">
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:15]: What is coloum?

File: app/components/public/add-to-calender.ts [hunk 7]
```diff
@@ -45,6 +45,7 @@ export default class AddToCalender extends Component<Args> {
     const { event } = this.args;
     const startTime = this.startsAt.utc().format('YYYY[-]MM[-]DDTHH[:]mm[:]SS[Z]');
     const endTime = this.endsAt.utc().format('YYYY[-]MM[-]DDTHH[:]mm[:]SS[Z]');
+
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:48]: Why?

**Comment by iamareebjamal** [comment]
Buttons aren't equal 
image

File: app/components/public/add-to-calender.hbs [hunk 8]
```diff
@@ -9,15 +9,43 @@
...
+          <UiDropdown class="ui floating dropdown button mt-4">
+            {{#if @event.sessions}}
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:17]: Wrong check. See how schedule menu item is shown

File: app/components/public/add-to-calender.ts (hunk 7 above)
Comment (Draco9421) [reply app/components/public/add-to-calender.ts:48]: oh I mistakenly added a line there.

File: app/components/public/add-to-calender.hbs [hunk 9]
```diff
@@ -9,15 +9,41 @@
...
-        {{#each this.calendarUrls as |calendar|}}
-          <a href={{calendar.url}} class="item" target="_blank" rel="noreferrer nofollow">
-            {{calendar.name}}
-          </a>
-        {{/each}}
+
+    <div class="ui grid mt-4">
+      <div class="eight wide column p-1">
+        <UiDropdown class="ui floating fluid dropdown button mb-2">
+          {{#if @event.sessions}}
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:16]: read previous review

Comment (Draco9421) [reply app/components/public/add-to-calender.hbs:16]: I didn't found any other way than that to make button equals.

File: app/components/public/add-to-calender.hbs (hunk 8 above)
Comment (Draco9421) [reply app/components/public/add-to-calender.hbs]: are you talking about these ?
Screenshot from 2020-12-24 23-08-11

**Review by daretobedifferent18 — COMMENTED** [review]
Screenshot from 2020-12-24 23-50-53
buttons text is not centered Pls fix this

File: app/components/public/add-to-calender.hbs (hunk 8 above)
Comment (iamareebjamal) [reply app/components/public/add-to-calender.hbs]: That's call for speakers, see schedule menu item

File: app/components/public/add-to-calender.hbs (hunk 9 above)
Comment (iamareebjamal) [reply app/components/public/add-to-calender.hbs:16]: https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803

**Comment by Draco9421** [comment]
> Screenshot from 2020-12-24 23-50-53
> buttons text is not centered Pls fix this

@daretobedifferent18 Please check now.

**Comment by Draco9421** [comment]
> Screenshot from 2020-12-24 23-50-53
> buttons text is not centered Pls fix this

@daretobedifferent18 Please check now.

**Comment by Draco9421** [comment]
Can you tell me where is schedule menu item like it's route or how it looks, screenshot will also work ?

**Comment by Draco9421** [comment]
@iamareebjamal please review it.

**Comment by iamareebjamal** [comment]
Already reviewed on 24 Dec https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r548761133

**Comment by Draco9421** [comment]
> Already reviewed on 24 Dec [#6088 (comment)](https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r548761133)

But I could not find where is schedule menu item , so please help me on that.

**Comment by iamareebjamal** [comment]
side menu

**Comment by Draco9421** [comment]
> side menu

?

**Comment by Draco9421** [comment]
Sir I still don't get it where is the problem in my current PR you can see that buttons are equal ? please explain me 


eq

**Comment by iamareebjamal** [comment]
https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803

**Comment by iamareebjamal** [comment]
You are changing random things while ignoring what I have commented on

**Comment by Draco9421** [comment]
> You are changing random things while ignoring what I have commented on

Sorry sir I tried to understand what you are trying to tell me in the comment , but I couldn't get through it.Please provide any image or screen shot of it so that I can understand it better.

**Comment by iamareebjamal** [comment]
I never talked about looks

https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803

I talked about logic

**Comment by Draco9421** [comment]
> I never talked about looks
> 
> [#6088 (comment)](https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803)
> 
> I talked about logic

Sir are you talking about the use of **if condition** or I have used some irrelevant class of semantic UI.

**Comment by iamareebjamal** [comment]
You cannot use @event.sessions

**Comment by Draco9421** [comment]
> You cannot use @event.sessions

But sir it was suggested to me by @daretobedifferent18 [#6088 (comment)](https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r546265765)

**Comment by iamareebjamal** [comment]
And after that, unsuggsted by me https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803 

For the 100th time

**Comment by Draco9421** [comment]
> And after that, unsuggsted by me [#6088 (comment)](https://github.com/fossasia/open-event-frontend/pull/6088#discussion_r547534803)
> 
> For the 100th time

Sir from last one month I was thinking about you are talking about the design of the button.

File: app/components/public/add-to-calender.hbs [hunk 10]
```diff
@@ -1,20 +1,20 @@
...
 
-    <div class="ui two item menu">
-      <div class="item">
-        <UiDropdown class="ui floating fluid dropdown text center aligned">
-          <i class="add to calendar icon"></i>
-          {{#if @event.sessions}}
+    <div class="ui grid mt-4">
+      <div class="eight wide column p-1">
+        <UiDropdown class="ui floating fluid dropdown button mb-2 text center aligned">
+          {{#if this.Session}}
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:11]: Won't work

For 20th time now, see how schedule menu item is shown

File: app/components/public/add-to-calender.hbs [hunk 11]
```diff
@@ -4,15 +4,40 @@
...
+        </UiDropdown>
       </div>
-    </UiDropdown>
+      <div class="eight wide column p-1">
+        {{#if (and this.isSessionPublished this.showSessions)}}    
+          <UiDropdown class="ui floating fluid dropdown button mb-2 text center aligned">
+            <i class="add to calendar icon"></i>
+            {{t 'Add Sessions'}}
+            <div class="menu">
+              {{#each this.SessioncalendarUrls as |calendar|}}
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.hbs:31]: Why is `SessioncalendarUrls` starting from a capital letter?

File: app/components/public/add-to-calender.ts [hunk 12]
```diff
@@ -31,6 +37,20 @@ export default class AddToCalender extends Component<Args> {
     return desc;
   }
 
+  constructor(owner:null, args:Args) {
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:40]: Fix formatting

File: app/components/public/add-to-calender.ts [hunk 13]
```diff
@@ -13,6 +15,10 @@ export default class AddToCalender extends Component<Args> {
 
   @service loader: any;
 
+  @tracked
+  showSessions : any;
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:19]: fix formatting

File: app/components/public/add-to-calender.ts [hunk 14]
```diff
@@ -13,6 +15,10 @@ export default class AddToCalender extends Component<Args> {
...
+
+
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:21]: extra line

File: app/components/public/add-to-calender.ts [hunk 15]
```diff
@@ -71,4 +91,12 @@ export default class AddToCalender extends Component<Args> {
     return [{ name: 'Google Calendar', url: this.googleUrl }, { name: 'iCal', url: this.iCalUrl }, { name: 'Yahoo', url: this.yahooUrl }, { name: 'Outlook', url: this.outlookUrl }];
   }
 
+  get SessioniCalUrl(): string {
```
Comment (iamareebjamal) [inline app/components/public/add-to-calender.ts:94]: Why capital? Is it class name?

I thought you got the idea from my last review? Or by seeing that no other method name startss from capital letter

Comment (Draco9421) [reply app/components/public/add-to-calender.ts]: oh sorry sir

**Comment by iamareebjamal** [comment]
OK, now it should work with Google Calendar as well

**Comment by mariobehling** [comment]
As the sidebar is a bit messed up, I updated the branch for testing.

**Comment by mariobehling** [comment]
Currently still appears like this for me here https://open-event-frontend-git-fork-draco9421-dev1.eventyay.now.sh/e/a315169f

Screenshot from 2021-01-31 23-43-52

**Comment by Draco9421** [comment]
> OK, now it should work with Google Calendar as well

Sir any idea that how can I add multiple sessions to google calendar ?

**Comment by Draco9421** [comment]
> Currently still appears like this for me here https://open-event-frontend-git-fork-draco9421-dev1.eventyay.now.sh/e/a315169f
> 
> Screenshot from 2021-01-31 23-43-52

Sir it might be because of CSS and the zoom level of the screen.

**Comment by iamareebjamal** [comment]
See how sched.io does it

**Comment by Draco9421** [comment]
Sir I tried adding it google calendar but it always shows this message **This email address isn't associated with an active Google Calendar account: https://open-event.dokku.fossasia.org/v1/events/607aec97.ics?include_sessions. Please check the email address and try again.**

**Comment by Draco9421** [comment]
Sir I couldn't find a way to work it with google calendar.

**Comment by iamareebjamal** [comment]
Checking

**Comment by iamareebjamal** [comment]
https://calendar.google.com/calendar/u/0/r?cid=webcal://api.eventyay.com/v1/events/a74665c3.ics?include_sessions

**Comment by mariobehling** [comment]
Thanks, please align the left side of the calendar widget with the same padding as all other widgets. The width of the widget needs to be the same as the other widgets.

Screenshot from 2021-02-10 06-57-58

**Comment by mariobehling** [comment]
Please reduce the padding in the buttons to keep the text in one line in different resolutions.

**Comment by mariobehling** [comment]
Screenshot from 2021-02-10 07-04-20

Same here: Left side is not aligned with other widgets.

Please expand the button, so all text is in one line.

**Comment by mariobehling** [comment]
The current implementation of the button has the width and position as expected.

Screenshot from 2021-02-10 07-39-39

**Comment by mariobehling** [comment]
I am updating the branch to see the latest changes reflected in that test deployment as well. Thanks.

**Comment by iamareebjamal** [comment]
Button height is messed up now

image

**Comment by Draco9421** [comment]
sorry sir I am not good at CSS but I will try my best.

**Comment by Draco9421** [comment]
@iamareebjamal and @mariobehling please review it.

**Comment by Draco9421** [comment]
@iamareebjamal and @mariobehling please review it.

**Pull request merged**